app/views/results/common/_context_menu.js.erb

```diff
@@ -1,113 +1,46 @@
 <script>
-  /*
-    This file contains the Javascript code necessary for the functionality
-    of the annotation context menu. For the annotation context menu to
-    function properly, these dependencies are necessary:
-
-    Context-Menu and library related:
-      * jquery.ui-contextmenu.min.js (..\app\assets\javascripts\)
-        * Depends on jQuery and jQuery-ui
-      * context_menu.css (..\app\assets\stylesheets\)
-
-    Context-Menu button functionality:
-      * _annotations.js.erb (..\app\views\results\common\)
-
-     */
   (function() {
-    var released_to_students = <%= @result.released_to_students %>;
+    var released_to_students = <%= result.released_to_students %>;
     if (released_to_students) return;
-    var menu_items = {
-      new_annotation: {
-        title: 'New Annotation',
-        cmd: 'new_annotation',
-        action: make_new_annotation,
-        disabled: true
-      },
-      delete_annotation: {
-        title: 'Delete Annotation',
-        cmd: "delete_annotation",
-        action: function(event, ui) {
-          var clicked_element = $(ui.target);
-          var annot_id = '';
-          var sub_file_id = $('#select_file_id').val();
-          if (annotation_type === ANNOTATION_TYPES.CODE) {
-            $.each(clicked_element[0].attributes, function(index, attr) {
-              if (attr.nodeName.toLowerCase()
-                      .indexOf('data-annotationid') != -1) {
-                annot_id = attr.value;
-                // Continue iteration in case of multiple annotations
-              }
-            });
-          } else {
-            annot_id = clicked_element.attr('id')
-                                      .replace('annotation_holder_', '');
-          }
-
-          if (annot_id !== null && annot_id.length !== 0 &&
-              sub_file_id !== null && sub_file_id.length !== 0) {
-            var query_str_params = [
-              ['id=' + annot_id],
-              ['submission_file_id=' + sub_file_id],
-              ['result_id=<%= @result.id %>'],
-              ['assignment_id=<%= @result.submission.grouping.assignment.id %>']
-            ];
-            $.ajax({
-              url: '<%= annotations_path %>?' + query_str_params.join('&'),
-              method: 'POST',
-              data: { _method: 'delete' },
-              dataType: 'script'
-            });
 
+    var annot_path = '<%= annotations_path %>';
+    var result_id = '<%= result.id %>';
+    var assignment_id = '<%= result.submission.grouping.assignment.id %>';
+    var file_dl_path = '<%= download_assignment_submission_result_path(
```

Comment:
Thanks for pointing this out, you're correct that we can't use the path helper here. (We *can* append `.erb` to the extension for arbitrary static Ruby code though, like internationalization.)

I was distracted by the fact that you're already passing in ids here, which is logically redundant. It reminds me that one of my wishlist is using [js-routes](https://github.com/railsware/js-routes).

Anyway, you don't have to change anything here.